MiniDB/PropertyChangedExtendedEventArgs.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MiniDB
+{
+    /// <summary>
+    /// An extension to PropertyChanged that includes the old and new values in addition to the changed properties
+    /// from https://stackoverflow.com/a/7742890 modified
+    /// </summary>
+    public class PropertyChangedExtendedEventArgs : PropertyChangedEventArgs
+    {
+        #region constructors
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PropertyChangedExtendedEventArgs"/> class.
+        /// </summary>
+        /// <param name="propertyName">The changed property</param>
+        /// <param name="oldValue">The old value</param>
+        /// <param name="newValue">The new value</param>
+        /// <param name="undoable">Whether this change is undoable or not</param>
+        public PropertyChangedExtendedEventArgs(string propertyName, object oldValue, object newValue, bool undoable = true)
```

Comment:
given places where you construct a new PropertyChangedExtenendEventArgs object based on an existing one, you may include a second constructor that takes in a string and a PropertyChangedExtendedEventArgs.